crates/turborepo-lib/src/run/watch.rs

```diff
@@ -213,8 +214,24 @@ impl WatchClient {
                     .build(handler, telemetry.clone())
                     .await?;
 
-                // Execute run
-                run.run().await?;
+                if run.has_persistent_tasks() {
+                    // Abort old run
+                    if let Some(run) = persistent_tasks_handle.take() {
+                        run.abort();
```

Comment:
Does this run synchronously? Do we need to `await` the handle after calling `abort`?

Reply:
So the actual cancelation [will likely happen at the next await point](https://docs.rs/tokio/latest/tokio/task/index.html#cancellation). We don't need to await the handle.